h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -444,35 +479,55 @@ public static MVStore open(String fileName) {
      */
     public synchronized <M extends MVMap<K, V>, K, V> M openMap(
             String name, MVMap.MapBuilder<M, K, V> builder) {
-        checkOpen();
-        String x = meta.get("name." + name);
-        int id;
-        long root;
+        int id = getMapId(name);
         M map;
-        if (x != null) {
-            id = DataUtils.parseHexInt(x);
-            @SuppressWarnings("unchecked")
-            M old = (M) maps.get(id);
-            if (old != null) {
-                return old;
-            }
-            map = builder.create();
-            String config = meta.get(MVMap.getMapKey(id));
-            String v = DataUtils.getFromMap(config, "createVersion");
-            map.init(this, id, v != null ? DataUtils.parseHexLong(v): 0);
-            root = getRootPos(meta, id);
+        if (id >= 0) {
+            map = openMap(id, builder);
         } else {
+            HashMap<String, Object> c = new HashMap<>();
             id = ++lastMapId;
-            map = builder.create();
-            map.init(this, id, currentVersion);
-            markMetaChanged();
-            x = Integer.toHexString(id);
+            c.put("id", id);
+            c.put("createVersion", currentVersion);
```

Comment:
Reintroduction of `HashMap` usage here is really needed? It looks like merge artifact. Some `HashMap` usages were removed in #952.

Reply:
Those two parameters were moved to builder, so they can be final. We may change builder interface as well.